File: mobile/iosApp/iosApp/UiIcons.swift
import Foundation

/// Maps semantic icon token names to SF Symbols (iOS).
/// See docs/ui-system.md — no shared cross-platform asset pack.
enum UiIcons {
    static func systemName(_ semanticName: String) -> String {
        switch semanticName {
        case UiTokens.Icon.calendar: return "calendar"
        case UiTokens.Icon.carpool: return "car"
        case UiTokens.Icon.family: return "person.3"
        case UiTokens.Icon.more: return "ellipsis.circle"
        case UiTokens.Icon.places: return "mappin.and.ellipse"
        case "icon.garage": return "door.garage.closed"
        case UiTokens.Icon.feeds: return "dot.radiowaves.up.forward"
        case UiTokens.Icon.signout: return "rectangle.portrait.and.arrow.right"
        case UiTokens.Icon.add: return "plus"
        case UiTokens.Icon.chevron: return "chevron.right"
        default:
            preconditionFailure("Unknown semantic icon: \(semanticName)")
        }
    }
}

File: web/src/components/uiIcons.ts
import type { LucideIcon } from "lucide-react"
import {
  Calendar,
  Car,
  ChevronRight,
  CircleEllipsis,
  LogOut,
  MapPin,
  Plus,
  Rss,
  Users,
} from "lucide-react"

/** Semantic icon names from design-tokens → Lucide (web). */
export const semanticIcons = {
  "icon.calendar": Calendar,
  "icon.carpool": Car,
  "icon.family": Users,
  "icon.more": CircleEllipsis,
  "icon.places": MapPin,
  "icon.feeds": Rss,
  "icon.signout": LogOut,
  "icon.add": Plus,
  "icon.chevron": ChevronRight,
} as const

export type SemanticIconName = keyof typeof semanticIcons

export function resolveSemanticIcon(name: SemanticIconName): LucideIcon {
  return semanticIcons[name]
}
